yt_dlp/extractor/audiodraft.py

```diff
@@ -0,0 +1,93 @@
+from .common import InfoExtractor
+from ..utils import int_or_none
+
+
+class AudiodraftBaseIE(InfoExtractor):
+    def _audiodraft_extract_from_id(self, player_entry_id):
+        data_json = self._download_json(
+            'https://www.audiodraft.com/scripts/general/player/getPlayerInfoNew.php', player_entry_id,
+            headers={
+                'Content-type': 'application/x-www-form-urlencoded; charset=UTF-8',
+                'X-Requested-With': 'XMLHttpRequest',
+            }, data=f'id={player_entry_id}'.encode('utf-8'))
+
+        return {
+            'id': str(data_json['entry_id']),
+            'title': data_json.get('entry_title'),
+            'url': data_json['path'],
+            'vcodec': 'none',
+            'ext': 'mp3',
+            'uploader': data_json.get('designer_name'),
+            'uploader_id': data_json.get('designer_id'),
+            'webpage_url': data_json.get('entry_url'),
+            'like_count': int_or_none(data_json.get('entry_likes')),
+            'average_rating': int_or_none(data_json.get('entry_rating')),
+        }
+
+
+class AudiodraftCustomIE(AudiodraftBaseIE):
+    IE_NAME = 'Audiodraft:custom'
+    _VALID_URL = r'https?://(?:[-\w]+)\.audiodraft\.com/entry/(?P<id>\d+)'
```

Comment:
Just to note: though it’s `https?://` here, none of the custom URLs I tried actually work via HTTPS in a browser. Who knows, maybe they will one day.